lstchain/io/dataselection.py

```diff
@@ -0,0 +1,248 @@
+from ctapipe.core import Component
+from ctapipe.core.traits import Int, Float, List
+from lstchain.reco.utils import filter_events
+
+import numpy as np
+import astropy.units as u
+from astropy.table import QTable
+from pyirf.binning import create_bins_per_decade  # , add_overflow_bins
+
+__all__ = ["DataSelection", "DataBinning"]
+
+
+class DataSelection(Component):
+    """
+    Collect various selection cuts to be applied for IRF production and
+    DL3 data reduction
+
+    Parameters for event filters will be combined in a dict so that the
+    filter_events() can be used.
+    """
+
+    intensity = List(
+        help="Range of intensity of event filter",
+        trait=Float(),
+        default_value=[0, np.inf],
+    ).tag(config=True)
+
+    length = List(
+        help="Range of length for event filter",
+        trait=Float(),
+        default_value=[0, np.inf],
+    ).tag(config=True)
+
+    width = List(
+        help="Range of width for event filter",
+        trait=Float(),
+        default_value=[0, np.inf],
+    ).tag(config=True)
+
+    r = List(
+        help="Range of r for event filter",
+        trait=Float(),
+        default_value=[0, 1],
+    ).tag(config=True)
+
+    wl = List(
+        help="Range of wl for event filter",
+        trait=Float(),
+        default_value=[0.01, 1],
+    ).tag(config=True)
+
+    leakage_intensity_width_2 = List(
+        help="Range for leakage_intensity_width_2 of event filter",
+        trait=Float(),
+        default_value=[0, 1],
+    ).tag(config=True)
+
+    fixed_gh_cut = Float(
+        help="Fixed selection cut for gh_score (gammaness)",
+        default_value=0.5,
+    ).tag(config=True)
+
+    fixed_theta_cut = Float(
+        help="Fixed selection cut for theta",
+        default_value=0.5,
+    ).tag(config=True)
+
+    fixed_source_fov_offset_cut = Float(
+        help="Fixed selection cut for source FoV offset",
+        default_value=3,
+    ).tag(config=True)
+
+    lst_tel_ids = List(
+        help="List of selected LST telescope ids",
+        trait=Int(),
+        default_value=[1],
+    ).tag(config=True)
+
+    def __init__(self, parent=None, **kwargs):
+        super().__init__(parent=parent, **kwargs)
+
+    def __call__(self, data):
```

Comment:
Call still does nothing

Reply:
It does not check. It logs debug message with that information. And I don't see why a class called `DataSelection` should have a `__call__` that only logs debug information.

`__call__` should either do the main "work" of the class or it is not needed.

Why did you add it like this?